### PR TITLE
Force a default of None to all Option fields without an explicit default

### PR DESCRIPTION
**Pull request by russellremple**
Play Json assumes None as the default for Option types, so None is serialized as missing from the output, and when data is missing on input, deserialization assigns None to these fields. upickle only processes defaults as specified explicitly. To be compatible deserializing JSON serialized with Play Json previously, WeePickle will automatically force a default of None to all Option fields without an explicit default.

@htmldoug @jducoeur @weisong-rallyhealth

### CODE REVIEW

File: weepickle/test/src/com/rallyhealth/weepickle/v1/StructTests.scala [hunk 1]
```diff
@@ -140,6 +140,18 @@ object StructTests extends TestSuite {
...
+//      implicit val r = WeePickle.macroR[Bar]
+//      assert(WeePickle.read[Bar]("""{}""") == Bar(None))
+//    }
+//
+//    test("use explicitly-provided default for Option types") {
+//      case class Bar(noDefault: Option[Int] = Some(1))
+//      implicit val r = WeePickle.macroR[Bar]
+//      assert(WeePickle.read[Bar]("""{}""") == Bar(Some(1)))
+//    }
+
```
Comment (jducoeur) [inline weepickle/test/src/com/rallyhealth/weepickle/v1/StructTests.scala]: Why are these commented out?  They seem pretty important to prove the concept.

Comment (jducoeur) [reply weepickle/test/src/com/rallyhealth/weepickle/v1/StructTests.scala]: (And immediately after saying this, I realized the commenting happened in a subsequent commit.)

Comment (russellremple) [reply weepickle/test/src/com/rallyhealth/weepickle/v1/StructTests.scala]: I thought they were causing build issues, but I was wrong. Latest commit should resolve it, and the tests are back (and actually compile now, which is a big plus!)

File: implicits/src/com/rallyhealth/weepickle/v1/implicits/internal/Macros.scala [hunk 2]
```diff
@@ -59,12 +59,21 @@ object Macros {
...
     }
 
-    def deriveDefaults(companion: c.Tree, hasDefaults: Seq[Boolean]): Seq[c.Tree] = {
+    /*
+     * Play Json assumes None as the default for Option types, so None is serialized as missing from the output,
+     * and when data is missing on input, deserialization assigns None to these fields. upickle only processes
+     * defaults as specified explicitly. To be compatible deserializing JSON serialized with Play Json previously,
+     * WeePickle will automatically force a default of None to all Option fields without an explicit default.
+     */
+    def deriveDefaults(companion: c.Tree, hasDefaults: Seq[Boolean], forceDefaultNone: Seq[Boolean]): Seq[c.Tree] = {
```
Comment (htmldoug) [inline implicits/src/com/rallyhealth/weepickle/v1/implicits/internal/Macros.scala]: Naming: "force" seems a bit strong. It's only used if a default isn't provided, right? I'd expect something like "force" to override the provided value.

**Pull request merged**